hnn_core/tests/test_gui.py

```diff
@@ -563,3 +564,42 @@ def add_child_decorated(self, to_add):
     # Check if the widgets are relinked, the selected index should be synced
     gui.tab_group_1.selected_index = 0
     assert gui.tab_group_2.selected_index == 0
+
+
+def test_gui_download_simulation():
```

Comment:
Could you add a test for csv upload in `test_gui_upload_data`?  That current test downloads test files from GitHub for the GUI testing. I don't think that's best practice as there is then a dependency on an external api. That will be change in another PR. 

For the scope of this PR you can add a test csv to `test/assets` for you test. A small/short file is preferred.

Reply:
Done. I created a test that uploads a csv file.